app/graphql/resolvers/update_user.rb

```diff
@@ -0,0 +1,53 @@
+class Resolvers::UpdateUser < Resolvers::MutationFunction
+  # arguments passed as "args"
+  argument :id, !types.ID
+  argument :first_name, types.String
+  argument :last_name, types.String
+  argument :email, types.String
+  argument :role, types.String
+  argument :profile_picture_b64, as: :attachment do
+      type types.String
+      description 'The base64 encoded bersion of the attatchment to upload.'
+  end
+
+  # return type from the mutation
+  type Types::UserType
+
+  # the mutation method
+   # _obj - is parent object, which in this case is nil
+  # args - are the arguments passed
+  # _ctx - is the GraphQL context (which would be discussed later)
+  def call(_obj, args, ctx)
+    if !Authentication::editor_is_valid(ctx) or !Authentication::admin_is_valid(ctx)
+      return GraphQL::ExecutionError.new("Invalid user token. Please log in.")
+    end
+    
+    @user = User.find(args["id"])
+
+    # Transaction so that we don't update a malformed article
+    User.transaction do
+      @user.first_name = args["first_name"] if args["first_name"]
+      @user.last_name = args["last_name"] if args["last_name"]
+      @user.email = args["email"] if args["email"]
+      @user.profile_picture = args["attachment"] if args["attachment"]
```

Comment:
Removing an image should be a possibility. So maybe allow nil to be passed to remove it. Since uploading pictures is relatively expensive, you can make an attachment of "" mean "keep the picture the same." If you take this approach, make sure to document it using the graphql doc stuff.

Reply:
Made an issue for this as well.